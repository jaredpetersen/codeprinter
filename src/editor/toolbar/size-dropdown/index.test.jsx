import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import SizeDropdown from './index';
import { DropdownItem } from 'reactstrap';

describe('Editor Toolbar SizeDropdown', () => {
  it('renders without crashing', () => {
    const sizes = [8, 9, 10, 11, 12];
    shallow(<SizeDropdown sizes={sizes} active={sizes[0]} onSelect={() => {}} />);
  });

  it('renders all of the sizes as dropdown options', () => {
    const sizes = [8, 9, 10, 11, 12];
    const tree = renderer.create(<SizeDropdown sizes={sizes} active={sizes[1]} onSelect={() => {}} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('calls the onSelect function on selecting a dropdown option', () => {
    const sizes = [8, 9, 10, 11, 12];
    const onSelect = jest.fn();
    const sizeDropdown = shallow(<SizeDropdown sizes={sizes} active={sizes[0]} onSelect={onSelect} />);

    sizeDropdown
      .find(DropdownItem)
      .last()
      .simulate('click');

    expect(onSelect).toHaveBeenCalledTimes(1);
  });
});
