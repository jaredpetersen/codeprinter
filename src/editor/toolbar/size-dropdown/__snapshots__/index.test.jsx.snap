// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor Toolbar SizeDropdown renders all of the sizes as dropdown options 1`] = `
<div
  className="dropdown"
  onKeyDown={[Function]}
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-label={null}
    className="dropdown-toggle-select btn btn-outline-dropdown-toggle-select btn-block"
    onClick={[Function]}
    style={
      Object {
        "textAlign": "left",
      }
    }
    type="button"
  >
    9
  </button>
  <div
    aria-hidden={true}
    className="dropdown-menu"
    role="menu"
    tabIndex="-1"
  >
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      8
    </button>
    <button
      className="dropdown-item active"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      9
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      10
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      11
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      12
    </button>
  </div>
</div>
`;
